fieldpath/set.go

```diff
@@ -206,6 +206,29 @@ func (s *Set) WithPrefix(pe PathElement) *Set {
 	return subset
 }
 
+// Leaves returns a set containing only the leaf paths
+// of a set.
+func (s *Set) Leaves() *Set {
+	members := MakePathElementSet(len(s.Members.members))
```

Comment:
This one is more likely to over-allocate since members are going to be filtered.

Reply:
reverted